diff --git a/src/daemon/root.rs b/src/daemon/root.rs
index e0204a8..4fe445b 100644
--- a/src/daemon/root.rs
+++ b/src/daemon/root.rs
@@ -121,8 +121,6 @@ impl DaemonContext for RootContext {
         self.state = state;
 
         let connection = daemon.get_connection();
-        let ftrace = Ftrace::init(&connection).await?;
-        daemon.add_service(ftrace);
 
         let ip = DeckService::init(connection);
         daemon.add_service(ip);
