diff --git a/data/platforms/jupiter.toml b/data/platforms/jupiter.toml
index 881ebb8..c8b7871 100644
--- a/data/platforms/jupiter.toml
+++ b/data/platforms/jupiter.toml
@@ -1,27 +1,27 @@
 [factory_reset.all]
-script  = "/usr/bin/steamos-reset-tool"
+script  = "@stubs@/bin/steamos-reset-tool"
 script_args = ["factory-reset", "--reset-all"]
 
 [factory_reset.os]
-script  = "/usr/bin/steamos-reset-tool"
+script  = "@stubs@/bin/steamos-reset-tool"
 script_args = ["factory-reset", "--reset-os"]
 
 [factory_reset.user]
-script  = "/usr/bin/steamos-reset-tool"
+script  = "@stubs@/bin/steamos-reset-tool"
 script_args = ["factory-reset", "--reset-user-data"]
 
 [update_bios]
-script = "/usr/bin/jupiter-biosupdate"
+script = "@steamDeckFirmware@/bin/jupiter-biosupdate"
 script_args = ["--auto"]
 
 [update_dock]
-script = "/usr/lib/jupiter-dock-updater/jupiter-dock-updater.sh"
+script = "@jupiterDockUpdaterBin@/jupiter-dock-updater/jupiter-dock-updater.sh"
 
 [storage.trim_devices]
-script = "/usr/lib/hwsupport/trim-devices.sh"
+script = "@hwsupport@/lib/hwsupport/trim-devices.sh"
 
 [storage.format_device]
-script = "/usr/lib/hwsupport/format-device.sh"
+script = "@hwsupport@/lib/hwsupport/format-device.sh"
 label_flag = "--label"
 device_flag = "--device"
 no_validate_flag = "--skip-validation"
diff --git a/data/system/com.steampowered.SteamOSManager1.service b/data/system/com.steampowered.SteamOSManager1.service
index 792ea8d..0088dc3 100644
--- a/data/system/com.steampowered.SteamOSManager1.service
+++ b/data/system/com.steampowered.SteamOSManager1.service
@@ -1,5 +1,5 @@
 [D-BUS Service]
 Name=com.steampowered.SteamOSManager1
-Exec=/usr/lib/steamos-manager -r
+Exec=@out@/lib/steamos-manager -r
 User=root
 SystemdService=steamos-manager.service
diff --git a/data/system/steamos-manager.service b/data/system/steamos-manager.service
index 1e21630..fda4ee6 100644
--- a/data/system/steamos-manager.service
+++ b/data/system/steamos-manager.service
@@ -7,7 +7,7 @@ After=steamos-log-submitter.service
 Type=dbus
 BusName=com.steampowered.SteamOSManager1
 Environment=RUST_LOG='INFO'
-ExecStart=/usr/lib/steamos-manager -r
+ExecStart=@out@/lib/steamos-manager -r
 Restart=on-failure
 RestartSec=1
 
diff --git a/data/user/com.steampowered.SteamOSManager1.service b/data/user/com.steampowered.SteamOSManager1.service
index fccfb36..f6a0d20 100644
--- a/data/user/com.steampowered.SteamOSManager1.service
+++ b/data/user/com.steampowered.SteamOSManager1.service
@@ -1,6 +1,6 @@
 [D-BUS Service]
 Name=com.steampowered.SteamOSManager1
-Exec=/usr/lib/steamos-manager
+Exec=@out@/lib/steamos-manager
 User=root
 SystemdService=steamos-manager.service
 
diff --git a/data/user/orca.service b/data/user/orca.service
index a378a66..ebe7313 100644
--- a/data/user/orca.service
+++ b/data/user/orca.service
@@ -3,7 +3,7 @@ Description=Orca Screen Reader
 
 [Service]
 Type=simple
-ExecStart=/usr/bin/orca
-ExecReload=/usr/bin/orca --replace
+ExecStart=@orca@/bin/orca
+ExecReload=@orca@/bin/orca --replace
 Restart=on-failure
 EnvironmentFile=%t/gamescope-environment
diff --git a/data/user/steamos-manager.service b/data/user/steamos-manager.service
index c147e40..d32e118 100644
--- a/data/user/steamos-manager.service
+++ b/data/user/steamos-manager.service
@@ -6,6 +6,6 @@ After=gamescope.service
 Type=dbus
 BusName=com.steampowered.SteamOSManager1
 Environment=RUST_LOG='INFO'
-ExecStart=/usr/lib/steamos-manager
+ExecStart=@out@/lib/steamos-manager
 Restart=on-failure
 RestartSec=1
diff --git a/src/daemon/root.rs b/src/daemon/root.rs
index e0204a8..82946bd 100644
--- a/src/daemon/root.rs
+++ b/src/daemon/root.rs
@@ -105,7 +105,7 @@ impl DaemonContext for RootContext {
     }
 
     fn system_config_path(&self) -> Result<PathBuf> {
-        Ok(path("/usr/share/steamos-manager/system.d"))
+        Ok(path("@out@/share/steamos-manager/system.d"))
     }
 
     fn state(&self) -> RootState {
diff --git a/src/daemon/user.rs b/src/daemon/user.rs
index c5db378..502db4f 100644
--- a/src/daemon/user.rs
+++ b/src/daemon/user.rs
@@ -67,7 +67,7 @@ impl DaemonContext for UserContext {
     }
 
     fn system_config_path(&self) -> Result<PathBuf> {
-        Ok(path("/usr/share/steamos-manager/user.d"))
+        Ok(path("@out@/share/steamos-manager/user.d"))
     }
 
     fn state(&self) -> UserState {
diff --git a/src/manager/root.rs b/src/manager/root.rs
index 9971811..ea0ac99 100644
--- a/src/manager/root.rs
+++ b/src/manager/root.rs
@@ -172,7 +172,7 @@ impl SteamOSManager {
         };
         for index in indices {
             let result = script_output(
-                "/usr/bin/steamos-polkit-helpers/jupiter-get-als-gain",
+                "@polkitHelpers@/bin/steamos-polkit-helpers/jupiter-get-als-gain",
                 &["-s", index],
             )
             .await;
diff --git a/src/platform.rs b/src/platform.rs
index ad24c78..29fc4ef 100644
--- a/src/platform.rs
+++ b/src/platform.rs
@@ -173,7 +173,7 @@ impl PlatformConfig {
             _ => return Ok(None),
         };
         let config = read_to_string(format!(
-            "/usr/share/steamos-manager/platforms/{platform}.toml"
+            "@out@/share/steamos-manager/platforms/{platform}.toml"
         ))
         .await?;
         Ok(Some(toml::from_str(config.as_ref())?))
diff --git a/src/wifi.rs b/src/wifi.rs
index 679b06f..e5041cc 100644
--- a/src/wifi.rs
+++ b/src/wifi.rs
@@ -32,19 +32,18 @@ use crate::{path, read_config_directory};
 
 const OVERRIDE_CONTENTS: &str = "[Service]
 ExecStart=
-ExecStart=/usr/lib/iwd/iwd -d
+ExecStart=@iwd@/libexec/iwd -d
 ";
 const OVERRIDE_FOLDER: &str = "/etc/systemd/system/iwd.service.d";
 const OVERRIDE_PATH: &str = "/etc/systemd/system/iwd.service.d/99-valve-override.conf";
 
 // Only use one path for output for now. If needed we can add a timestamp later
 // to have multiple files, etc.
-const TRACE_CMD_PATH: &str = "/usr/bin/trace-cmd";
+const TRACE_CMD_PATH: &str = "@traceCmd@/bin/trace-cmd";
 
 const MIN_BUFFER_SIZE: u32 = 100;
 
 const WIFI_BACKEND_PATHS: &[&str] = &[
-    "/usr/lib/NetworkManager/conf.d",
     "/etc/NetworkManager/conf.d",
 ];
 
@@ -228,11 +227,11 @@ pub(crate) async fn get_wifi_backend() -> Result<WifiBackend> {
 }
 
 pub(crate) async fn set_wifi_backend(backend: WifiBackend) -> Result<()> {
-    run_script("/usr/bin/steamos-wifi-set-backend", &[backend.to_string()]).await
+    run_script("@stubs@/bin/steamos-wifi-set-backend", &[backend.to_string()]).await
 }
 
 pub(crate) async fn list_wifi_interfaces() -> Result<Vec<String>> {
-    let output = script_output("/usr/bin/iw", &["dev"]).await?;
+    let output = script_output("@iw@/bin/iw", &["dev"]).await?;
     Ok(output
         .lines()
         .filter_map(|line| match line.trim().split_once(' ') {
@@ -246,7 +245,7 @@ pub(crate) async fn get_wifi_power_management_state() -> Result<WifiPowerManagem
     let mut found_any = false;
     for iface in list_wifi_interfaces().await? {
         let output =
-            script_output("/usr/bin/iw", &["dev", iface.as_str(), "get", "power_save"]).await?;
+            script_output("@iw@/bin/iw", &["dev", iface.as_str(), "get", "power_save"]).await?;
         for line in output.lines() {
             match line.trim() {
                 "Power save: on" => return Ok(WifiPowerManagement::Enabled),
@@ -267,7 +266,7 @@ pub(crate) async fn set_wifi_power_management_state(state: WifiPowerManagement)
 
     for iface in list_wifi_interfaces().await? {
         run_script(
-            "/usr/bin/iw",
+            "@iw@/bin/iw",
             &["dev", iface.as_str(), "set", "power_save", state],
         )
         .await
